x/emissions/keeper/score_utils.go

```diff
@@ -6,77 +6,161 @@ import (
 }
 
-// Return low score and index among all inferences
-func GetLowScoreFromAllInferences(
+// Update lowest score from new inferer addresses set
+func UpdateLowestScoreFromInfererAddresses(
 	ctx context.Context,
 	k *Keeper,
 	topicId TopicId,
-	inferences types.Inferences,
-) (lowScore types.Score, lowScoreIndex int, err error) {
-	lowScoreIndex = 0
-	lowScore, err = k.GetInfererScoreEma(ctx, topicId, inferences.Inferences[0].Inferer)
+	infererAddresses []string,
+	addedInferer string,
+	removedInfererAddress string,
+) error {
+	// remove inferer from the list
+	for i, address := range infererAddresses {
+		if address == removedInfererAddress {
+			infererAddresses = append(infererAddresses[:i], infererAddresses[i+1:]...)
+			break
+		}
+	}
+	// add new inferer to the list
+	infererAddresses = append(infererAddresses, addedInferer)
+	lowScore, err := k.GetInfererScoreEma(ctx, topicId, infererAddresses[0])
 	if err != nil {
-		return types.Score{}, lowScoreIndex, err
+		return err
 	}
-	for index, extInference := range inferences.Inferences {
-		extScore, err := k.GetInfererScoreEma(ctx, topicId, extInference.Inferer)
+	for _, address := range infererAddresses {
+		score, err := k.GetInfererScoreEma(ctx, topicId, address)
+		if err != nil {
+			return err
+		}
+		if lowScore.Score.Gt(score.Score) {
+			lowScore = score
+		}
+	}
+	return k.SetLowestInfererScoreEma(ctx, topicId, lowScore)
+}
+
+// Get lowest score from all inferers
+func GetLowestScoreFromAllInferers(
+	ctx context.Context,
+	k *Keeper,
+	topicId TopicId,
+	infererAddresses []string,
+) (lowScore types.Score, err error) {
+	for i, address := range infererAddresses {
+		score, err := k.GetInfererScoreEma(ctx, topicId, address)
 		if err != nil {
 			continue
 		}
-		if lowScore.Score.Gt(extScore.Score) {
-			lowScore = extScore
-			lowScoreIndex = index
+		if lowScore.Score.Gt(score.Score) || i == 0 {
+			lowScore = score
 		}
 	}
-	return lowScore, lowScoreIndex, nil
+	return lowScore, nil
 }
 
-// Return low score and index among all forecasts
-func GetLowScoreFromAllForecasts(
+// Update lowest score from new forecaster addresses set
+func UpdateLowestScoreFromForecasterAddresses(
 	ctx context.Context,
 	k *Keeper,
 	topicId TopicId,
-	forecasts types.Forecasts,
-) (lowScore types.Score, lowScoreIndex int, err error) {
-	lowScoreIndex = 0
-	lowScore, err = k.GetForecasterScoreEma(ctx, topicId, forecasts.Forecasts[0].Forecaster)
+	forecasterAddresses []string,
+	addedForecaster string,
+	removedForecasterAddress string,
+) error {
+	// remove forecaster from the list
+	for i, address := range forecasterAddresses {
+		if address == removedForecasterAddress {
+			forecasterAddresses = append(forecasterAddresses[:i], forecasterAddresses[i+1:]...)
+			break
+		}
+	}
+	// add new forecaster to the list
+	forecasterAddresses = append(forecasterAddresses, addedForecaster)
+	lowScore, err := k.GetForecasterScoreEma(ctx, topicId, forecasterAddresses[0])
```

Comment:
This call and the loop below it repeat the functionality of `GetLowestScoreFromAllForecasters()`, so I think you should call it here. Same for the pairs of analogous functions of other actors.

Reply:
good catch!